src/Nethermind/Nethermind.Serialization.Rlp/ReceiptMessageDecoder.cs

```diff
@@ -58,14 +57,15 @@ public TxReceipt Decode(RlpStream rlpStream, RlpBehaviors rlpBehaviors = RlpBeha
             txReceipt.Bloom = rlpStream.DecodeBloom();
 
             int lastCheck = rlpStream.ReadSequenceLength() + rlpStream.Position;
-            List<LogEntry> logEntries = new List<LogEntry>();
 
-            while (rlpStream.Position < lastCheck)
+            int numberOfReceipts = rlpStream.ReadNumberOfItemsRemaining(lastCheck);
```

Comment:
here - we can count number of items remaining and construct a correct length array immediately